exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +60,465 @@ type BTCMarkets struct {
+			}
+			price, err = strconv.ParseFloat(temp[i].Bids[y][0], 64)
+			if err != nil {
+				return orderbooks, err
+			}
+			tempOB.Bids = append(tempOB.Bids, OBData{Price: price, Volume: volume})
+		}
+		orderbooks = append(orderbooks, tempOB)
 	}
+	return orderbooks, nil
+}
 
-	if !resp.Success {
-		return nil, errors.New(resp.ErrorMessage)
-	}
+// GetServerTime gets time from btcmarkets
+func (b *BTCMarkets) GetServerTime() (TimeResp, error) {
+	var resp TimeResp
+	return resp, b.SendHTTPRequest(btcMarketsAPIURL+btcMarketsAPIVersion+btcMarketsGetTime,
+		&resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetAccountBalance returns the full account balance
+func (b *BTCMarkets) GetAccountBalance() ([]AccountData, error) {
+	var resp []AccountData
+	return resp,
+		b.SendAuthenticatedRequest(http.MethodGet,
+			btcMarketsAccountBalance,
+			nil,
+			&resp)
+}
 
-		for j := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[j].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[j].Volume /= common.SatoshisPerBTC
-		}
-	}
+// GetTradingFees returns trading fees for all pairs based on trading activity
+func (b *BTCMarkets) GetTradingFees() (TradingFeeResponse, error) {
+	var resp TradingFeeResponse
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		btcMarketsTradingFees,
+		nil,
+		&resp)
+}
 
-	return resp.Orders, nil
+// GetTradeHistory returns trade history
+func (b *BTCMarkets) GetTradeHistory(marketID, orderID, before, after, limit string) ([]TradeHistoryData, error) {
+	var resp []TradeHistoryData
+	req := make(map[string]interface{})
+	if marketID != "" {
+		req["marketId"] = marketID
+	}
+	if orderID != "" {
+		req["orderId"] = orderID
+	}
+	if before != "" {
+		req["before"] = before
+	}
+	if after != "" {
+		req["after"] = after
+	}
+	if limit != "" {
+		req["limit"] = limit
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTradeHistory, req, &resp)
 }
 
-// GetOrderDetail returns order information an a specific order
-// orderID - example "1337"
-func (b *BTCMarkets) GetOrderDetail(orderID []int64) ([]Order, error) {
-	type OrderDetail struct {
-		OrderIDs []int64 `json:"orderIds"`
+// NewOrder requests a new order and returns an ID
+func (b *BTCMarkets) NewOrder(marketID string, price, amount float64, orderType, side string, triggerPrice,
+	targetAmount float64, timeInForce string, postOnly bool, selfTrade, clientOrderID string) (OrderData, error) {
+	var resp OrderData
+	req := make(map[string]interface{})
+	req["marketId"] = marketID
+	req["price"] = strconv.FormatFloat(price, 'f', -1, 64)
+	req["amount"] = strconv.FormatFloat(amount, 'f', -1, 64)
+	req["type"] = orderType
+	req["side"] = side
+	if orderType == "Stop Limit" || orderType == "Take Profit" || orderType == "Stop" {
```

Comment:
Would be more user friendly if you create these as vars, convert to lower case and do the if check on them